python/ray/data/tests/test_filter.py

```diff
@@ -271,14 +321,18 @@ def test_filter_with_invalid_expression(ray_start_regular_shared, tmp_path):
 )
 def test_filter_with_predicate_expressions(
```

Comment:
Out of scope for this PR, but I think if we're going to have like 20+ parametrizations, we should refactor this as a unit test (no `ray.init`)